### ISSUE TITLE
System auto-generate of numerical attribute does not work

### ISSUE THREAD

**Issue by AkubaD** [HRHR-project/e-registry-app#24]
This has never worked for our eRegistry instance. This is the uid of the attribute I056EjniPUi
This other attribute (uid dsDPcUTQ60R) is what was used for Palestine. Using that has not worked either.
In the data entry page for creating a new tracked entity, the system generated ID is required (whether you set it to be mandatory or not), but nothing is auto-generated and filled in. So the system forces you to put in a number, because it will not create the tracked entity with a blank required field. It does check to ensure that the number is unique, but the auto-generate functionality does not work
Please see the settings below and let us know if it's because we are doing anything wrong with the configurations, or if it is a bug

![image](https://user-images.githubusercontent.com/36447512/36425027-b91d514c-1645-11e8-94fe-b87d28a0d680.png)

**Comment by Bekkalizer** [comment]
This is not supported in e-registry apps. Would have to be ported to be supported.

**Comment by zepto5** [comment]
Support for auto-generated values added in commit: 6acfe78

**Comment by AkubaD** [comment]
@zepto5 @Bekkalizer @lukasewvd 
We didn't use the System generated ID in the first phase of rollout, We are now using it for the first time since it was implemented in March and I get an error when I tried to create an attribute with a unique number generated with a pattern.
According to the instructions I saw online for DHIS2, to generate a pattern, you put in a number of zeros representing the number of digits you want. I put in 6 zeros for a 6 digit unique system generated ID. 
When I clicked to save the attribute, I got the message: "Failed to parse pattern '00000'. The  expression is invalid"
Attribute ID I056EjniPUi
![image](https://user-images.githubusercontent.com/36447512/39304121-ebf0afd8-4958-11e8-9247-dc697d1daa41.png)
